src/ln/channelmanager.rs

```diff
@@ -685,10 +693,22 @@ impl ChannelManager {
 						Ok(funding_msg) => {
 							(chan, funding_msg.0, funding_msg.1)
 						},
-						Err(_e) => {
-							//TODO: Push e to pendingevents
+						Err(e) => {
+							println!("Got error handling message: {}!", e.err);
+							if let Some(action) = e.msg {
+								match action {
+									msgs::ErrorAction::DisconnectPeer {} => {
+										add_pending_event!(events::Event::DisconnectPeer {
```

Comment:
Please dont take both the pending_events lock and the channel_stat lock at the same time here. I'm probably overparanoid about deadlocks, but we dont have any automated analysis to detect them so best to take the perf hit of just using a temporary vector to ensure they cant happen.